app/services/idv/financials_step.rb

```diff
@@ -1,32 +1,44 @@
 module Idv
   class FinancialsStep < Step
-    def complete?
-      idv_form.finance_type && idv_session.financials_confirmation.try(:success?) ? true : false
+    def submit
+      if complete?
+        @success = true
+        idv_session.financials_confirmation = true
```

Comment:
Before, `idv_session.financials_confirmation` was relied upon for 2 things: to see if the vendor validation was successful, and to pass in the errors to analytics. Now, the errors are fetched directly from `FinancialsValidator`. The session never needs to be involved. So then the only thing that was left was to find a simple way to know for sure if the confirmation passed or not so that a controller can redirect appropriately. I settled on the boolean assignment.

By having the FinancialsStep class return a hash that includes the errors, we can easily display them to the user via the controller, whereas before, the controller did not have access to the errors directly, which is why you had to store them in the session.